cmd/kubeadm/app/phases/certs/certlist.go

```diff
@@ -57,28 +58,57 @@ func (k *KubeadmCert) CreateFromCA(ic *kubeadmapi.InitConfiguration, caCert *x50
 	if err != nil {
 		return fmt.Errorf("couldn't create %q certificate: %v", k.Name, err)
 	}
-	cert, key, err := pkiutil.NewCertAndKey(caCert, caKey, *cfg)
+	cert, key, err := pkiutil.NewCertAndKey(caCert, caKey, cfg)
 	if err != nil {
 		return err
 	}
-	writeCertificateAuthorithyFilesIfNotExist(
+	writeCertificateFilesIfNotExist(
 		ic.CertificatesDir,
 		k.BaseName,
+		caCert,
 		cert,
 		key,
 	)
 
 	return nil
 }
 
+// CreateAsCA creates a certificate authority, writing the files to disk and also returning the created CA so it can be used to sign child certs.
+func (k *KubeadmCert) CreateAsCA(ic *kubeadmapi.InitConfiguration) (*x509.Certificate, *rsa.PrivateKey, error) {
+	cfg, err := k.GetConfig(ic)
+	if err != nil {
+		return nil, nil, fmt.Errorf("couldn't get configuration for %q CA certificate: %v", k.Name, err)
```

Comment:
Please use errors.Wrap on error propagation.

Reply:
We already use fmt.Errorf everywhere, at some point I'd like to just remove them all in one fell swoop. It seems like it might even be automateable